Add video and steps to vector search page (#9532)
opensearch-project/documentation-website: _vector-search/index.md

OpenSearch [vector search]({{site.url}}{{site.baseurl}}/vector-search/getting-started/vector-search-basics/) provides a complete vector database solution for building efficient AI applications. Store and search vector embeddings alongside your existing data, making it easy to implement semantic search, retrieval-augmented generation (RAG), recommendation systems, and other AI-powered applications.

## Overview

> Watch this video to learn about key vector search features in OpenSearch and see a step-by-step demo of using OpenSearch as a vector database.

Author: ```suggestion
Watch this video to learn about key vector search features in OpenSearch and discover how to use OpenSearch as a vector database through a step-by-step demo.
```

> The following video introduces key vector search features in OpenSearch and provides a step-by-step demo of using OpenSearch as a vector database.

Author: ```suggestion
Watch this video to learn about key vector search features in OpenSearch and see a step-by-step demo of using OpenSearch as a vector database.
```

{% include youtube-player.html id='oX0HMAztP8E' %}

To follow the demo, use these steps.

<details markdown="block">
  <summary>
    Steps
  </summary>
  {: .fs-5 .fw-700}

### Prerequisites
{:.no_toc} 

Download the sample data for this demo:

```bash
wget https://amazon-pqa.s3.amazonaws.com/amazon_pqa_headsets.json
```
{% include copy.html %}

Prepare data for bulk indexing into OpenSearch:

```bash
head -n 5000 amazon_pqa_headsets.json |  awk '{ print "{\"index\":{\"_index\":\"neural_search_pqa\"}}"; print;}'  > neural_search_amazon_pqa_headsets.json
```
{% include copy.html %}

> Enable running ML models on data nodes (not recommended for production environments):

Reviewer: ```suggestion
Enable running machine learning (ML) models on data nodes (not recommended for production environments):
```

```json
PUT /_cluster/settings
{
  "persistent": {
    "plugins.ml_commons.only_run_on_ml_node": false
  }
}
```
{% include copy-curl.html %}

### Step 1: Register and deploy a model
{:.no_toc} 

> Register and deploy a machine learning (ML) model provided by OpenSearch:

Reviewer: ```suggestion
Register and deploy an ML model provided by OpenSearch:
```

### Step 2: Create an ingest pipeline 
{:.no_toc} 

Create an ingest pipeline that will generate vector embeddings from text:

```json
PUT _ingest/pipeline/nlp-index-pipeline
{
  "processors" : [
    {
      "text_embedding": {
        "model_id": "<model_id>",
        "field_map": {
          "question_text": "question_vector"
        }
      }
    }
  ]
}
```
{% include copy-curl.html %}

Test the ingest pipeline:

```json
POST /_plugins/_ml/_predict/text_embedding/<model_id>
{
  "text_docs":[ "what does the package contain?"],
  "return_number": true,
  "target_response": ["sentence_embedding"]
}
```
{% include copy-curl.html %}

### Step 3: Create an index
{:.no_toc} 

Create a vector index and set the default ingest pipeline to the ingest pipeline created in the previous step:

```json
PUT /neural_search_pqa
{
  "settings": {
    "index.knn": true,
    "default_pipeline": "nlp-index-pipeline"
  },
  "mappings": {
    "properties": {
      "question_vector": {
        "type": "knn_vector",
        "dimension": 768
      }
    }
  }
}
```
{% include copy-curl.html %}

### Step 4: Ingest data
{:.no_toc} 

> Ingest the data you prepared in Prerequisites:

Reviewer: Either "Ingest the data you prepared in the [Prerequisites](link) section" or "Ingest the data you prepared as part of the [Prerequisites](link) for this tutorial" (include link to section).

> Ingest the data you prepared in Prerequisites:

Author: ```suggestion
Ingest the data you prepared in the [Prerequisites](#prerequisites) section:
```

```bash
curl -XPOST -u "<username>:<password>" -k https://localhost:9200/_bulk --data-binary @neural_search_amazon_pqa_headsets.json  -H 'Content-Type: application/json'
```
{% include copy.html %}

If you're not running the Security plugin, omit the username and password:

```bash
curl -XPOST http://localhost:9200/_bulk --data-binary @neural_search_amazon_pqa_headsets.json  -H 'Content-Type: application/json'
```
{% include copy.html %}

Test the vector generation:

```json
GET /neural_search_pqa/_search
```
{% include copy-curl.html %}

### Step 5: Search the data
{:.no_toc} 

> Now, search the data using various search methods.

Reviewer: ```suggestion
Now search the data using the following search methods.
```
